pkg/graph/resolvers/system_intake_actions.go

```diff
@@ -628,3 +628,200 @@ func CreateSystemIntakeActionNotITGovRequest(
 	}
 	return intake, nil
 }
+
+// UpdateLCID is used to update the LCID on a system intake.
+func UpdateLCID(
+	ctx context.Context,
+	store *storage.Store,
+	fetchUserInfo func(context.Context, string) (*models.UserInfo, error),
+	input model.SystemIntakeUpdateLCIDInput,
+) (*models.SystemIntake, error) {
+	adminEUAID := appcontext.Principal(ctx).ID()
+
+	adminUserInfo, err := fetchUserInfo(ctx, adminEUAID)
+	if err != nil {
+		return nil, err
+	}
+
+	intake, err := store.FetchSystemIntakeByID(ctx, input.SystemIntakeID)
+	if err != nil || intake == nil {
+		return nil, err
+	}
+	err = lcidactions.IsLCIDValidToUpdate(intake)
+	if err != nil {
+		return nil, err
+	}
+	// input.Reason //TODO: The reason field will be retained in the DB in a future ticket
+
+	// save action (including additional info for email, if any)
```

Comment:
Is there any particular reason to have the code for saving the action first, before actually updating the intake? I was pretty confused when first reading this, I wasn't sure if the intake was actually going to get updated and saved, especially because the convention we've been following is to update the intake and _then_ start saving everything. It might be necessary to call `GetConfirmLCIDAction()` first before updating `intake`, but even then, starting up the `errGroup` can wait, and we can add an explanatory comment about why `action` is being created early in the function.

Reply:
Because this action data relies on a pointer to the intake, there was a data race condition when the action was in the `errgroup`. Moving the  action declaration out of the go routine corrected the race condition.

Also, because this action is acting as a sort of audit we need the previous intake values. (This doesn't happen to the same extent in the other actions). 

I don't think it matters that the action is saved before the intake, since this should happen asynchronously. I'll add a comment.

Reply:
I can see the reasoning for calling `action := lcidactions.GetUpdateLCIDAction()` here. That said, if that gets done first, I don't _think_ there should be a pointer to the intake needed inside the errgroup? Once `intake` gets dereferenced as part of the call to `GetUpdateLCIDAction()`, the data processed by that function should be separate from the `intake` value that gets modified later in `UpdateLCID()`. 

As for which is saved first, the end result won't change, but I think the code reads a bit easier (and more like other actions) if `errGroup` is initialized and the action is saved inside an `errGroup.Go()` call later in the function, _after_ updating fields on the intake, as part of saving everything to the DB.

Reply:
Fair enough, I moved the saving action til after the intake fields are modified.

Reply:
Looks good now.